upload_components/upload.sh

```diff
@@ -2,7 +2,7 @@
 
 IFS=';' read -ra DIRECTORIES <<<"${COMPONENTS_DIRECTORIES:-.}"
 NAMESPACE=${COMPONENTS_NAMESPACE:-espressif}
-if [ -n "$SKIP_PRE_RELEASE" ]; then
+if [[ "$SKIP_PRE_RELEASE" == "true" ]]; then
```

Comment:
This potentially can be a breaking change, because before  setting it to `1`, would also work, 
However, it's unlikely that anyone used this option so far, so it's ok,
The only thing, I would suggest making the comparison case-insensitive, because in the readme it is spelled as `False`/`True` (bash should be new enough on ubuntu-latest for it) 

```suggestion
if [[ "${SKIP_PRE_RELEASE,,}" == "true" ]]; then
```

Reply:
Interestingly, if this value is set to `True` in yaml becomes `true` in bash, but I agree that case-insensitive match won't hurt.

Alternatively, we can just fix it in the docs to `enable if non-empty` which won't be a breaking change.